NearestExonJB.pm

```diff
@@ -105,20 +103,18 @@ sub run {
   my $vf = $tva->base_variation_feature;
   my $trv = $tva->base_transcript_variation;
 
-  if(!exists($self->{_cache}) || !exists($self->{_cache}->{$trv->transcript_stable_id})) {
-    $self->{config}->{ea} = $self->{config}->{reg}->get_adaptor($self->{config}->{species}, $self->{config}->{core_type}, 'Exon');
-    $self->{ea} ||= $self->{config}->{ea};
-    die("ERROR: Could not get exon adaptor;\n") unless $self->{ea};
+  my $loc_string = sprintf("%s:%s-%i-%i", $trv->transcript_stable_id, $vf->{chr} || $vf->seq_region_name, $vf->{start}, $vf->{end});
 
-    my @exons = @{$trv->transcript->get_all_Exons};
+  if(!exists($self->{_cache}) || !exists($self->{_cache}->{$loc_string})) {
+    my @exons = @{$trv->_exons};
```

Comment:
You don't need to access exon number to get the overlapped exon.

I'd suggest instead:

```
my @exons = $trv->_overlapped_exons;

if(scalar @exons == 1) {
  $exon = $exons[0];
  ## rest of code as you have it
}
else {
  ## either no exon overlap (@exons == 0) or overlaps more than 1 (@exons > 1)
  ## bail out and return {} ???
}
```